File: LeetCode/LeetCode_Easy/TypeScript/ValidParentheses.ts
function isValid(s: string): boolean {
    const stack: Array<string> = [];

    for (let i = 0; i < s.length; i++) {
        if (s[i] === '(') { stack.push(')'); continue; }

        if (s[i] === '[') { stack.push(']'); continue; }

        if (s[i] === '{') { stack.push('}'); continue; }

        if (stack.length <= 0 || s[i] !== stack.pop()) { return false; }
    }

    return stack.length === 0;
};

File: LeetCode/LeetCode_Easy/C++/ValidParentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> stack;

        for (int i = 0; i < s.size(); i++) {
            if (s[i] == '(') { stack.push(')'); continue; }

            if (s[i] == '{') { stack.push('}'); continue; }

            if (s[i] == '[') { stack.push(']'); continue; }

            if (stack.empty() || stack.top() != s[i]) return false;
        
            stack.pop();
        }
        return stack.empty();
    }
};